### PR TITLE
Update Cryptographic Signatures to use recommended hashing algorithm

### LINKED ISSUE #30272: Update hashing algorithm in code samples to match recommendations
Explanation recommends using SHA256; code samples all still use SHA1.

Additionally:
"The Modulus property is set to a value called rsaModulus" is not helpful; it's repeating the code. Provide explanation of what modulus/exponent are and where they come from regarding the creation of a new RSA properties object.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d1e9012c-5727-23eb-7d9f-24f144e2545f
* Version Independent ID: 33e3a24e-0384-eb47-d74a-b4bd820b4969
* Content: [Cryptographic Signatures](https://docs.microsoft.com/en-us/dotnet/standard/security/cryptographic-signatures)
* Content Source: [docs/standard/security/cryptographic-signatures.md](https://github.com/dotnet/docs/blob/main/docs/standard/security/cryptographic-signatures.md)
* Product: **dotnet**
* Technology: **dotnet-security**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Use recommended `SHA256` hashing algorithm, instead of `SHA1`.

Fixes #30272

## Internal preview

- [Cryptographic Signatures](https://review.docs.microsoft.com/en-us/dotnet/standard/security/cryptographic-signatures?branch=pr-en-us-30535)

### CODE REVIEW

File: docs/standard/security/cryptographic-signatures.md [hunk 1]
```diff
@@ -30,129 +30,83 @@ This topic explains how to generate and verify digital signatures using classes
...
-      //signedHashValue.
-      signedHashValue = rsaFormatter.CreateSignature(hashValue);
-   }
+    signedHash = rsaFormatter.CreateSignature(hash);
 }
+
+// The sharedParameters, hash, and signedHash are used to later verify the signature.
 ```
 
 ## Verifying Signatures
```
Comment (gewarren) [inline docs/standard/security/cryptographic-signatures.md:91]:
```suggestion
## Verify a signature
```

File: docs/standard/security/cryptographic-signatures.md [hunk 2]
```diff
@@ -30,129 +30,83 @@ This topic explains how to generate and verify digital signatures using classes
 
 ## Generating Signatures
```
Comment (gewarren) [inline docs/standard/security/cryptographic-signatures.md:31]:
```suggestion
## Generate a signature
```

File: docs/standard/security/cryptographic-signatures.md [hunk 3]
```diff
@@ -30,129 +30,83 @@ This topic explains how to generate and verify digital signatures using classes
...
 
 - The public key of the party that signed the data.
-
 - The digital signature.
-
 - The data that was signed.
-
 - The hash algorithm used by the signer.
 
 To verify a signature signed by the <xref:System.Security.Cryptography.RSAPKCS1SignatureFormatter> class, use the <xref:System.Security.Cryptography.RSAPKCS1SignatureDeformatter> class. The <xref:System.Security.Cryptography.RSAPKCS1SignatureDeformatter> class must be supplied the public key of the signer. For RSA, you will need the values of the modulus and the exponent to specify the public key. (The party that generated the public/private key pair should provide these values.) First create an <xref:System.Security.Cryptography.RSA> object to hold the public key that will verify the signature, and then initialize an <xref:System.Security.Cryptography.RSAParameters> structure to the modulus and exponent values that specify the public key.
```
Comment (gewarren) [inline docs/standard/security/cryptographic-signatures.md:100]: Why are we removing the modulus/exponent example? They're still mentioned on line 100.

Comment (IEvangelist) [reply docs/standard/security/cryptographic-signatures.md:100]: Those are the only two fields that are required on Windows, the others are optional. Instead, I'm trying to highlight that the creation of the signature should instead export its parameters, and those parameters should be imported later by the verifying RSA. The original example didn't show where those values came from, or how to correctly provide them. Does that make sense?

**Review by gewarren — APPROVED** [review]

**Pull request merged**